libraries/AP_Math/crc.cpp

```diff
@@ -599,13 +599,19 @@ uint8_t parity(uint8_t byte)
     return p;
 }
 
-// sums the bytes in the supplied buffer, returns that sum mod 256
-// (i.e. shoved into a uint8_t)
-uint8_t crc_sum_of_bytes(uint8_t *data, uint16_t count)
+// sums the bytes in the supplied buffer, returns that sum mod 0xFFFF
+uint16_t crc_sum_of_bytes_16(const uint8_t *data, uint16_t count)
 {
-    uint8_t ret = 0;
+    uint16_t ret = 0;
     for (uint32_t i=0; i<count; i++) {
         ret += data[i];
     }
     return ret;
 }
+
+// sums the bytes in the supplied buffer, returns that sum mod 256
+// (i.e. shoved into a uint8_t)
+uint8_t crc_sum_of_bytes(const uint8_t *data, uint16_t count)
```

Comment:
How is this different than the already merged https://github.com/ArduPilot/ardupilot/blob/60ceaec9012a5e86148f2ed77232bb6c252e94ce/libraries/AP_Math/crc.cpp#L602-L611

`crc_sum_of_bytes`

Reply:
this new one is 16 bit. I made the 8 bit one use the 16 bit one and mask